### PR TITLE
[CI] post land-blocking test result on PR

### PR DESCRIPTION
**Pull request by sausagee**
## Motivation
The land-blocking test is run at land time in the auto branch. When the
test fails or performance regression is detected, the relevant results
need to be surfaced in the original pull request such that the author
can follow up.

## Test Plan
- Mock 1 - Cluster test fails and doesn't generate test report
https://github.com/sausagee/libra/runs/442190978?check_suite_focus=true
image

- Mock 2 - Cluster test passes and generates test report  
https://github.com/sausagee/libra/runs/442177298?check_suite_focus=true
image

- Mock 3 - Cluster test passes but the commit is not found in any open pull request. 
https://github.com/sausagee/libra/runs/442462648?check_suite_focus=true
image

### CODE REVIEW

**Comment by andll** [comment]
Should we just extract `text` part? Json can get pretty big eventually

File: .github/workflows/land-blocking.yml [hunk 1]
```diff
@@ -11,19 +11,72 @@ jobs:
...
+          docker/build-aws.sh --build-all --version $LIBRA_GIT_REV --addl_tags canary
+      - name: Launch cluster test
+        run: |
+          JUMPHOST=${{secrets.CLUSTER_TEST_JUMPHOST}} \
+          ./scripts/cti \
+            --marker land \
+            --tag dev_$LIBRA_GIT_REV \
+            --report report.json \
+            --run bench \
+            -- --duration 60
```
Comment (andll) [inline .github/workflows/land-blocking.yml]: I thought a little bit more about it - maybe lets remove `-- --duration 60` for now? I could make separate PR for that, but since you are already touching this piece, do you think you could remove it?

Comment (sausagee) [reply .github/workflows/land-blocking.yml]: SG.  Will remove it.

**Comment by sausagee** [comment]
> Should we just extract `text` part? Json can get pretty big eventually

I think it depends on what is useful for review.  I will extract the text part.  Maybe you can think about what are the relevant metrics to add in the future?

File: .github/workflows/land-blocking.yml [hunk 2]
```diff
@@ -11,19 +11,72 @@ jobs:
...
+      - name: Post test results on PR
+        if: always()
+        uses: actions/github-script@0.4.0
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            should_fail = false;
```
Comment (metajack) [inline .github/workflows/land-blocking.yml]: Can we use `let`? So `let should_fail = false;`.

Comment (metajack) [reply .github/workflows/land-blocking.yml]: Also, are we missing a `let body = "";`? Where is `body` initialized?

File: .github/workflows/land-blocking.yml [hunk 3]
```diff
@@ -11,19 +11,72 @@ jobs:
...
+            });
+            const pulls = await github.pulls.list(
+              {
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                state: "open",
+                sort: "updated",
+              }
+            );
+            pr_num = 0;
```
Comment (metajack) [inline .github/workflows/land-blocking.yml]: Use `let`.

File: .github/workflows/land-blocking.yml [hunk 4]
```diff
@@ -11,19 +11,72 @@ jobs:
...
+            -- --duration 60
+      - name: Post test results on PR
+        if: always()
+        uses: actions/github-script@0.4.0
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            should_fail = false;
+            const fs = require('fs');
+            fs.readFile('report.json', 'utf-8', (err, data) => {
```
Comment (metajack) [inline .github/workflows/land-blocking.yml]: Please use `await` here. Since Node 11 fs.readFile returns promises, and the script action uses Node 12. See https://stackoverflow.com/a/56821924. That will let you get rid of the `should_fail` stuff too.

File: .github/workflows/land-blocking.yml (hunk 2 above)
Comment (sausagee) [reply .github/workflows/land-blocking.yml]: Good call. Was following the github-script examples.  Will update.

**Comment by andll** [comment]
Hiding json under `<details>` seem like a good idea

File: .github/workflows/land-blocking.yml (hunk 4 above)
Comment (sausagee) [reply .github/workflows/land-blocking.yml]: The `should_fail` is for separate failure cases (coming in the next few diffs)
- cluster test fails
- cluster test passes but TPS regresses x %
- cluster test passes but latency regresses y %
etc

**Comment by andll** [comment]
Since I don't really understand the language that you are using in this script, I will leave it to @metajack for final approve, but otherwise this looks good to me!

File: .github/workflows/land-blocking.yml (hunk 1 above)
Comment (sausagee) [reply .github/workflows/land-blocking.yml]: Fxied.

File: .github/workflows/land-blocking.yml (hunk 2 above)
Comment (sausagee) [reply .github/workflows/land-blocking.yml]: Fixed.

File: .github/workflows/land-blocking.yml (hunk 3 above)
Comment (sausagee) [reply .github/workflows/land-blocking.yml]: Fixed.

File: .github/workflows/land-blocking.yml (hunk 4 above)
Comment (sausagee) [reply .github/workflows/land-blocking.yml]: Updated but kept should_fail, as we will need it for the next phase (fail the workflow if there is perf regression).

File: .github/workflows/land-blocking.yml [hunk 5]
```diff
@@ -11,19 +11,80 @@ jobs:
...
+              return;
+            }
+            // Read and check cluster test results
+            let should_fail = false;
+            let body;
+            const fsp = require('fs').promises;
+            try {
+              data = await fsp.readFile('report.json', 'utf-8');
+              var result = JSON.parse(data);
+              // TODO - set P/F based on metrics TPS, latency
```
Comment (sausagee) [inline .github/workflows/land-blocking.yml]: @andll Maybe you can start thinking about this?  What threshold to use and where to store them.
